codec-http2/src/main/java/io/netty/handler/codec/http2/DecoratingHttp2FrameWriter.java

```diff
@@ -102,6 +102,12 @@ public ChannelFuture writeFrame(ChannelHandlerContext ctx, byte frameType, int s
         return delegate.writeFrame(ctx, frameType, streamId, flags, payload, promise);
     }
 
+    @Override
+    public ChannelFuture writeFrame(ChannelHandlerContext ctx, byte frameType, int streamId, short flags,
```

Comment:
My opinion is not super strong, but I don't see much value in increasing public API scope with the only motivation to reduce allocations, because instances of `Http2Flags` can be cached. There are not many variations of different flags. 

Other than that, lgtm after @bryce-anderson comments are addressed.

Reply:
That's actually a fair point ...

Reply:
The problem is when using it with http2 extensions, where a custom frame can have many flags

Reply:
```java
        if (msg instanceof CustFrame) {
            final CustFrame frame = (CustFrame) msg;
            final byte[] data = frame.getData();
            final byte mappedHttp2FrameType = spdyFrameType2Http2FrameTypeMapping(frame.getType().getCode());
            encoder().writeFrame(
                ctx,
                mappedHttp2FrameType, 
                frame.stream().id(),
                new Http2Flags(frame.getFlags()), //TODO avoid boxing
                Unpooled.wrappedBuffer(data),
                promise
            );
            return;
        }
```

@bryce-anderson I want to avoid the boxing in my code.